core/lib/multivm/src/vm_instance.rs

```diff
@@ -328,3 +328,11 @@ impl<S: ReadStorage, Tr: Tracer + Default + 'static> FastVmInstance<S, Tr> {
         Self::Shadowed(ShadowedFastVm::new(l1_batch_env, system_env, storage_view))
     }
 }
+
+/// Checks whether the protocol version is supported by the fast VM.
+pub fn is_supported_by_fast_vm(protocol_version: ProtocolVersionId) -> bool {
```

Comment:
We have a lot of such switches inside protocol version, probably it's better to keep it there and do not spread it across the code base

Reply:
I feel that in this case, it is justified. The existing switches tell something inherent about protocol versions, while this one tells something about VM implementations (which are downstream of protocol versions; protocol versions should not be aware of how switching between implementations is implemented).